### PR TITLE
Support turning off the interactive mode at individual request level

### PR DESCRIPTION
**Pull request by harryi3t**

### CODE REVIEW

File: test/integration/auth-methods/digest.test.js [hunk 1]
```diff
@@ -87,6 +89,69 @@ describe('digest auth', function () {
...
+                        name: 'DigestAuth',
+                        request: {
+                            url: 'https://postman-echo.com/digest-auth',
+                            auth: {
+                                type: 'digest',
+                                digest: {
+                                    algorithm: 'MD5',
+                                    username: '{{uname}}',
+                                    password: '{{pass}}',
+                                    nonInteractive: true
```
Comment (kunagpal) [inline test/integration/auth-methods/digest.test.js]: 1. The schema will have to be updated to reflect this as well.
2. Add this test for all other applicable auth modes as well.

Comment (harryi3t) [reply test/integration/auth-methods/digest.test.js]: 1. will do
2. already done (digest auth and NTLM auth only)

File: lib/authorizer/digest.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
 var crypto = require('crypto-js'),
+
+    NON_INTERACTIVE = 'nonInteractive',
```
Comment (kamalaknn) [inline lib/authorizer/digest.js]: IMO, now that this flag is not generic but specific to individual auths, it should not be called `nonInteractive`.

Comment (harryi3t) [reply lib/authorizer/digest.js]: @kamalaknn what should it be called?

Comment (shamasis) [reply lib/authorizer/digest.js]: These should be like @kamalaknn says - specific to each auth
digest may have some terminology in the RFC

File: lib/authorizer/ntlm.js [hunk 3]
```diff
@@ -83,6 +84,11 @@ module.exports = {
      * @param {AuthHandlerInterface~authPostHookCallback} done
      */
     post: function (auth, response, done) {
+        // bail out if interactive mode is disabled
```
Comment (shamasis) [inline lib/authorizer/ntlm.js]: again, this has to be NTLM specific property name

File: lib/authorizer/digest.js (hunk 2 above)
Comment (kamalaknn) [reply lib/authorizer/digest.js]: The RFC also uses the retry in its wording.

```
The client is expected to retry the request, passing an Authorization header line
```

It should be `disableRetry{something}`.

Comment (harryi3t) [reply lib/authorizer/digest.js]: @kamalaknn Same is true for NTLM also
https://msdn.microsoft.com/en-us/library/cc237544.aspx
image

**Pull request merged**